### PR TITLE
Support reading .npz files in ZIP64

### PR DESCRIPTION
**Pull request by zhihaoy**
- refactored the code to parse with packed structs
- the files for testing are not exactly ZIP64 files, but we only read local headers anyway.
- unified some error messages

### CODE REVIEW

File: include/xtensor-io/xnpz.hpp [hunk 1]
```diff
@@ -39,6 +39,98 @@ namespace xt
 {
     using namespace std::string_literals;
 
+    namespace detail
+    {
+#pragma pack(push, 2)
```
Comment (JohanMabille) [inline include/xtensor-io/xnpz.hpp]: Why is this required? Besides, this is Windows specific stuff so it should be guarded.

File: include/xtensor-io/xnpz.hpp [hunk 2]
```diff
@@ -39,6 +39,98 @@ namespace xt
...
+        };
+#pragma pack(pop)
+
+        struct extensible_data_field
+        {
+            uint16_t header_id;
+            uint16_t data_size;
+        };
+
+#pragma pack(push, 4)
```
Comment (JohanMabille) [inline include/xtensor-io/xnpz.hpp]: Same question here

**Comment by SylvainCorlay** [comment]
Thank you for the contribution! @zhihaoy

File: include/xtensor-io/xnpz.hpp (hunk 2 above)
Comment (wolfv) [reply include/xtensor-io/xnpz.hpp]: I think we remove the padding so the binary schema is the same as in the file.

Comment (wolfv) [reply include/xtensor-io/xnpz.hpp]: so that you can reinterpret_cast the first couple of bytes and see if everything falls into place ... that's how I understood it -- also the tests are green so I assume it works on OS X and Linux (and MSVC has docs for it as well).

Comment (JohanMabille) [reply include/xtensor-io/xnpz.hpp]: This pragma is ignored on Linux and OSX, so having the tests green probably means that this padding is not really required.

Comment (zhihaoy) [reply include/xtensor-io/xnpz.hpp]: Some information about these packed structs: I need to remove the padding between data members and the padding in the end because ZIP isn't designed to take alignment into account.  The `alignas` specifier can only increase alignment, not remove or reduce alignment.

https://godbolt.org/z/CX22QZ (removing the pragma causes `sizeof` to increase)

There is still a problem remaining: The `crc32` member and the two subsequent members are unaligned given their type.  It doesn't work on old ARM chips.  But it may not worth a fix (for example, split the local header into two halves) before we see the first bug report.

Comment (JohanMabille) [reply include/xtensor-io/xnpz.hpp]: Thanks @zhihaoy for the explanation. I agree with you regarding the remaining members, let's wait and see if someone complains about them.
